### ISSUE TITLE
Minecraft Mod issues

### ISSUE THREAD

**Issue by zsawyer** [thorwe/CrossTalk#14]
I have just tested CrossTalk 1.5.0 with Minecraft 1.5.1 and the [MumbleLink mod](https://sourceforge.net/projects/modmumblelink/). 

Sadly it does not work as intended. 
- there is no attenuation (getting quieter as the avatars move away)
- the directional is only an abrupt switching between left ear and right ear (stereo headset) (the other tester [7.1 headset] said it was either only on the right ear or both)
- further testing showed that both ears were working for me when I turned around but when the other tester walked around me in a circle the direction only sometimes worked abit (abit meaning that there was never an audible transition from left to right or similar)
- testing with Mumble running in parallel worked perfectly
- MumblePAHelper reported good parameters (z axis is being zeroed out)

**Comment by PvtPirate** [comment]
I tested the beta with MumbleLink-mod:
the position update will stay as an issue, but i know you cannot change that.
the attenuation now works, though it is difficult to set up properly if you have no clue about how much 1dB is.
it was bugged in the other version, inverting attenuation: it went more silent when getting closer and had full volume when you walk away past the maximum distance - but apparently you fixed that in the current beta.

for easier setup, maybe you could change how attenuation variables are set by the user:
min distance [m]
volume [%] at min distance [m]
max distance [m]
volume [%] at max distance [m]

past max distance, everything will be at this volume %

**Comment by zsawyer** [comment]
Thanks for the update and feedback guys. I am sad to not have been able to test it myself so far. 

About the sent interval problem: Would it be possible to interpolate the movement and instead of "spamming" the server with updates have the engine calculate an approximate location based on the last few packets of data. This would stop it from sounding like the audio source is zapping around and make it more fluently. 

Also the problem that the sent interval value's minimum is 0.1 seconds still persists.

**Comment by PvtPirate** [comment]
zsawyer, that would not change anything, as the coordinates are being sent in this way
<whatever game or game plugin>  -->  CrossTalk  -->  TeamSpeak
and even if Crosstalk would calculate an estimated next position to be able to send a more fluent strem of coords to TS, this still doesn't allow TS to accept more coordinate updates per minute.
This is the bottleneck and it cannot be changed by thorwe, but only by a serveradmin with corresponding userrights.

**Comment by thorwe** [comment]
I guess he meant the receiving side would calculate an estimate for the time inbetween incoming info.
I'm not sure about that in regards to bang for the buck (the later being my limited time).

If there are indeed server admins that have a problem with adjusting the flood points, it might be worthy to rather go full and create an optional / second stream via WebSockets, e.g. to a node js server with authentication (maybe meteor) that'd distribute those.

Note that creating an additional ticket for single items is indeed welcome, in general threads I just tend to forget. I'll create one for the minimum update value, however it'll be 0.02s as the mumble link protocol is at a fixed 50 times per second rate for updates.

**Comment by zsawyer** [comment]
True, the interpolation should happen on the receiver side due to the problem described by pvtpirate. 
I am still curious how ACRE does it since they do not have this issue (although flood rate is recommended to be turned up for that afaik). 

Allowing higher flood rates is a reasonable solution. I understand that spending time to fix something which originates at a different endpoint is not very efficient. I just was afraid (sorry for not checking) that TS's had a floodrate max of 0.1 second so that is why I was not sure if it is even possible to go lower. But you seem quite certain so that will suffice.

**Comment by thorwe** [comment]
Server Admins can even simply disable the flood detection for a certain user group if they're trusted, so technically there's no hard limit there that ain't exceedable.
It's just a question of the config, sadly my only choice is to set rather high default values to not have users freaking out with default server settings.
